api/migrations/20200812121443-createCertificateAmendments.js

```diff
@@ -0,0 +1,200 @@
+
+    for (const amendment of certificateAmendments) {
+      let flavourRecordResult = null;
+      let masterRecordResult = null;
+      let flavourRecord = null;
+
+      // Check if there is an LNG flavour.
+      if (amendment._flavourRecords && amendment._flavourRecords.length) {
+        // Epic certificates should only have a single LNG flavour, but lets confirm.
+        flavourRecord = await nrpti.findOne({ _id: ObjectId(amendment._flavourRecords[0]) });
+
+        if (flavourRecord._schemaName.includes('LNG')) {
+          flavourRecordResult = await createLngFlavour(nrpti, flavourRecord);
+        } else {
+          console.log(`Master record ${amendment._id} had a flavour that wasn't LNG. Check flavour record ${flavourRecord._id}`);
+        }
+      }
+
+      // Create the master record.
+      masterRecordResult = await createMaster(nrpti, amendment, flavourRecordResult.insertedId);
+
+      // Delete the originals.
+      if (masterRecordResult) {
+        await deleteRecord(nrpti, amendment._id);
+      }
+
+      if (flavourRecordResult) {
+        await deleteRecord(nrpti, flavourRecord._id);
+      }
+    }
+
+  } catch (err) {
+    console.log('Error:', err);
+  }
+
+  console.log('Done.');
+  mClient.close()
+};
+
+exports.down = function(db) {
+  return null;
+};
+
+exports._meta = {
+  "version": 1
+};
+
+async function createLngFlavour(nrptiCollection, flavourRecord) {
+  if (!nrptiCollection || !flavourRecord) {
+    return;
+  }
+
+  const transformedRecord = {
+    _schemaName: 'CertificateAmendmentLNG',
+    _epicProjectId: flavourRecord._epicProjectId || '',
+    _sourceRefId: flavourRecord.sourceRefId || '',
+    _epicMilestoneId: flavourRecord._epicMilestoneId || '',
+
+    mineGuid: flavourRecord.mineGuid || '',
+    read: flavourRecord.read || [],
+    write: flavourRecord.write || [],
+    recordName: flavourRecord.recordName || '',
+    recordType: 'CertificateAmendment',
+    recordSubtype: '',
+    dateIssued: flavourRecord.dateIssued || null,
+    issuingAgency: flavourRecord.issuingAgency || '',
+    legislationDescription: flavourRecord.legislationDescription || '',
+    projectName: flavourRecord.projectName || '',
+    location: flavourRecord.location || '',
+    centroid: flavourRecord.centroid || [],
+    documents: flavourRecord.documents || [],
+    description: flavourRecord.description || '',
+    dateAdded: flavourRecord.dateAdded || null,
+    dateUpdated: flavourRecord.dateUpdated || null,
+    addedBy: flavourRecord.addedBy || '',
+    updatedBy: flavourRecord.updatedBy || '',
+    sourceDateAdded: flavourRecord.sourceDateAdded || null,
+    sourceDateUpdated: flavourRecord.sourceDateUpdated || null,
+    sourceSystemRef: 'epic',
+    datePublished: flavourRecord.datePublished || null,
+    isLngPublished: flavourRecord.isLngPublished || false,
+    isBcmiPublished: flavourRecord.isBcmiPublished || false,
+
+    issuedTo: {
+      write: ['sysadmin'],
```

Comment:
Should this be just sysadmin or also lng admin?

Reply:
I wrote these to match the mongoose model defaults.  I will add them in here.